maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -3114,9 +3114,12 @@ protected void warnIfIllegalFailOnFlakeCount() throws MojoFailureException
 
     private void printDefaultSeedIfNecessary()
     {
-        if ( getRunOrderRandomSeed() == null && getRunOrder().equals( RunOrder.RANDOM.name() ) )
+        if ( getRunOrder().equals( RunOrder.RANDOM.name() ) )
         {
-            setRunOrderRandomSeed( System.nanoTime() );
+            if ( getRunOrderRandomSeed() == null )
+            {
+                setRunOrderRandomSeed( System.nanoTime() );
```

Comment:
Personally I would expect some warning that the seed is missing.
If random, then yes, we should use random and prevent from any cohesion between the neighbouring runs.

Reply:
Let's see the implementation, because the impl has much better opportunities to use real random number if this parameter is NULL.

Reply:
`runOrderRandomSeed`  is never NULLm if is not provided it will be set to `nanoTime`.

Currently is printed only when is not given in configuration, change introduce to print always - also if is given by configuration.

Reply:
@slawekjaranowski runOrder can be other than random and so `DefaultRunOrderCalculator` may obtain null seed.
@delanym see my comment https://github.com/apache/maven-surefire/pull/505#discussion_r844543780